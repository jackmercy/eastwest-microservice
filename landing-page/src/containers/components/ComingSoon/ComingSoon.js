import { memo } from 'react';
import { useIntl } from 'react-intl';

import Content from '../Content';

import useStyles from './comingSoon.styles';

const ComingSoon = () => {
  const intl = useIntl();
  const classes = useStyles();

  return (
    <>
      <div className={classes.teaser} data-test="coming-soon-teaser-image">
        <Content className={classes.teaserContent}>
          <h1
            className={classes.teaserTitle}
            data-test="coming-soon-teaser-title"
          >
            {intl.formatMessage({ id: 'COMING_SOON.TEASER.TITLE' })}
          </h1>
        </Content>
      </div>
      <Content className={classes.content}>
        <img
          alt="coming soon"
          src="/static/images/coming-soon/cover.png"
          loading="lazy"
          className={classes.image}
          data-test="coming-soon-cover-image"
        />
        <h1 className={classes.title} data-test="coming-soon-title">
          {intl.formatMessage({ id: 'COMING_SOON.TITLE' })}
        </h1>
        <p className={classes.description} data-test="coming-soon-description">
          {intl.formatMessage({ id: 'COMING_SOON.DESCRIPTION' })}
        </p>
      </Content>
    </>
  );
};
export default memo(ComingSoon);
