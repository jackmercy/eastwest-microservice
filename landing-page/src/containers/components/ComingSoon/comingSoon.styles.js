import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';
import COLORS from '../../../constants/COLORS';

export default createUseStyles({
  teaser: {
    width: '100%',
    height: rem(280),
    boxSizing: 'border-box',
    backgroundImage: 'url(/static/images/coming-soon/teaser.png)',
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'right',
  },
  teaserContent: {
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    height: '100%',
    color: COLORS.textWhite,
    boxSizing: 'border-box',
  },
  teaserTitle: {
    margin: 0,
    fontWeight: 900,
    fontSize: rem(72),
    lineHeight: rem(88),
  },
  content: {
    boxSizing: 'border-box',
    textAlign: 'center',
    paddingTop: rem(32),
    paddingBottom: rem(192),
  },
  image: {
    width: '100%',
    display: 'inline-block',
    maxWidth: rem(512),
  },
  title: {
    fontWeight: 'bold',
    fontSize: rem(32),
    lineHeight: rem(48),
    marginTop: rem(16),
    marginBottom: 0,
  },
  description: {
    fontSize: rem(16),
    lineHeight: rem(32),
    marginTop: rem(8),
    marginBottom: 0,
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    teaser: {
      height: rem(184),
      backgroundImage: 'url(/static/images/coming-soon/teaser-mobile.png)',
    },
    teaserTitle: { fontSize: rem(40), lineHeight: rem(48) },
    content: { paddingBottom: rem(100) },
    image: { maxWidth: rem(300) },
    title: { fontSize: rem(18), lineHeight: rem(28) },
    description: { fontSize: rem(14), lineHeight: rem(20) },
  },
});
