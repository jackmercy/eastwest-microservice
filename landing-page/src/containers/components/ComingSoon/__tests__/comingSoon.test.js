import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ComingSoon from '../ComingSoon';

describe('Testing ComingSoon', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<ComingSoon />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<ComingSoon />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a teaser image', () => {
    expect(wrapper.find('[data-test="coming-soon-teaser-image"]')).toHaveLength(
      1,
    );
  });

  it('should have a teaser title', () => {
    expect(wrapper.find('[data-test="coming-soon-teaser-title"]')).toHaveLength(
      1,
    );
  });

  it('should have a cover image', () => {
    expect(wrapper.find('[data-test="coming-soon-cover-image"]')).toHaveLength(
      1,
    );
  });

  it('should have a title', () => {
    expect(wrapper.find('[data-test="coming-soon-title"]')).toHaveLength(1);
  });

  it('should have a description', () => {
    expect(wrapper.find('[data-test="coming-soon-description"]')).toHaveLength(
      1,
    );
  });
});
