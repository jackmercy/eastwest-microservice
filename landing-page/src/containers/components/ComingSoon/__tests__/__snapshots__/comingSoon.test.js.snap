// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ComingSoon should render match snapshot 1`] = `
<Fragment>
  <div
    className="teaser-0-2-1"
    data-test="coming-soon-teaser-image"
  >
    <Memo(Content)
      className="teaserContent-0-2-2"
    >
      <h1
        className="teaserTitle-0-2-3"
        data-test="coming-soon-teaser-title"
      >
        Coming soon
      </h1>
    </Memo(Content)>
  </div>
  <Memo(Content)
    className="content-0-2-4"
  >
    <img
      alt="coming soon"
      className="image-0-2-5"
      data-test="coming-soon-cover-image"
      loading="lazy"
      src="/static/images/coming-soon/cover.png"
    />
    <h1
      className="title-0-2-6"
      data-test="coming-soon-title"
    >
      Coming soon
    </h1>
    <p
      className="description-0-2-7"
      data-test="coming-soon-description"
    >
      This page is under construction
    </p>
  </Memo(Content)>
</Fragment>
`;
